Starcraft2Client.py

```diff
@@ -340,9 +340,10 @@ def build_mission_table(self, dt):
 
             def mission_callback(self, button):
                 if not self.launching:
-                    mission_id: int = list(self.mission_id_to_button.values()).index(button)
-                    self.ctx.play_mission(list(self.mission_id_to_button)
-                                          [mission_id])
+                    button_id: int = list(self.mission_id_to_button.values()).index(button)
+                    mission_id: int = list(self.mission_id_to_button)[button_id]
+
```

Comment:
```suggestion
                    mission_id: int = next(k for k, v in self.mission_id_to_button.items() if v == button)
```
maybe? would the two list() calls